commons/servers/handlers.ts

```diff
@@ -16,17 +16,17 @@ export function validateSignatureHandler(
   signerDataBuilder: (body: any) => SignerData = (b) => b,
   signatureValidator: SignatureValidator = Authenticator.validateSignature
 ): RequestHandler {
-  return async (req: Request, res: Response, next: NextFunction) => {
+  return (req: Request, res: Response, next: NextFunction) => {
     const signerData = signerDataBuilder(req.body)
 
-    await validateSignature(
+    validateSignature(
       signerData,
       `${messageToSignBuilder(req.body)}${signerData.timestamp}`,
       next,
       (message) => res.status(401).send({ status: 'unauthorized', message }),
       (signer) => authorizedSignerPredicate(signer, req.body),
       networkOrProvider,
       signatureValidator
-    )
+    ).catch(next)
```

Comment:
Shouldn't this respond with an error code? Does this work like that?

Reply:
the signature of next receives an error. `next(error?: any)` if no error is provided, it assumes a successful middleware, otherwise a special code path handles the error and sends the 500 to the users